### ISSUE TITLE
Display save folder in saved game dialog

### ISSUE THREAD

**Issue by DanRStevens** [OutpostUniverse/OPHD#929]
There's a common question of where saved games are stored on disk. It's a complicated looking path, and hard to remember. Maybe we should display the path as part of the saved game dialog. Then people would know where to find them. In particular, it makes it easier for people to upload saved games demonstrating a problem.

**Comment by belgianguy** [comment]
The path will still be quite complex, another idea would be to have a `savegames` folder next to the EXE, that would clear up a lot of confusion I think. As I naively will always check the EXE directory first if I am looking for a specific file.

**Comment by ldicker83** [comment]
> The path will still be quite complex, another idea would be to have a `savegames` folder next to the EXE, that would clear up a lot of confusion I think. As I naively will always check the EXE directory first if I am looking for a specific file.

This was how it was originally done but to be conformant with current Windows best practices for applications and to be consistent with *nix based systems (Linux, macOS, etc.), we simplified the calls in the Filesystem code.

I don't really see this as a problem that needs to be solved in the application. We can add it as a question to the FAQ.

Another option is to save it in the users Documents directory under Windows. Some games do this which still remains conformant with best practices but would require some platform dependent changes in NAS2D::Filesystem

**Comment by DanRStevens** [comment]
Yes, the reason for the strange path is that executable files are often stored in a part of the filesystem owned by a more privileged user, so normal users would have read-only access. That makes it impossible to store saved game files there. If you're using a local folder install, which might be typical with unzipping a distribution package and just running it, then yes, it would be possible then, but not in general.

For many applications, when you save a file, a file dialog box generally pops up that shows the path and allows you to change it. It think it would be fairly reasonable to just show the path, and not offer a way to change it. We could just add a label over the file list box, and show the current save path there.

I'm not so fond of the documentation approach, since people are still left searching for the info they need, and may not even know where to look. It's likely people will still end up asking, or just not bother with providing saved games because they can't be bothered to go looking. If you put the info right in the user interface though, then they have it available right when they're creating or loading the saved game. Plus, with users trained to check file location from file dialog boxes, this is the most obvious place for them to look.

If we really want to go the extra mile, we could even provide a shortcut button to popup the local file browser at the correct folder. I think there was similar code to popup a help or about page somewhere.

**Comment by ldicker83** [comment]
> For many applications, when you save a file, a file dialog box generally pops up that shows the path and allows you to change it. It think it would be fairly reasonable to just show the path, and not offer a way to change it. We could just add a label over the file list box, and show the current save path there.
> 
> I'm not so fond of the documentation approach, since people are still left searching for the info they need, and may not even know where to look. It's likely people will still end up asking, or just not bother with providing saved games because they can't be bothered to go looking. If you put the info right in the user interface though, then they have it available right when they're creating or loading the saved game. Plus, with users trained to check file location from file dialog boxes, this is the most obvious place for them to look.

Users are going to ask regardless of what labels we provide or what documentation is provided. That's the nature of users. Give up before searching or trying to do anything for themselves. Not a complaint, just a fact of the user experience.

> If we really want to go the extra mile, we could even provide a shortcut button to popup the local file browser at the correct folder. I think there was similar code to popup a help or about page somewhere.

I think this would be a better approach. We'd definitely need platform specific code here for this so we'll want to explore it but I'm not opposed to offering this functionality. It will break full-screen modes but since we're not using full-screen at the moment anyway I'm not worried about it.

For macOS this could be done as part of the system menu. Not sure if we want to get that far into the platform-specific code but that's an option as well.

**Comment by Brett208** [comment]
I agree that the current location is difficult to find. I would steer clear of storing the saved game next to the executable on Windows as that is not a best practice and could cause security features to interfere with operation.

To make it worse, the current save directory is in a directory chain that is hidden by default, which makes it even more difficult for someone not literate in Windows file explorer to find it.

An alternative for an easier storage path on Windows could be to place saved games and config settings in Documents/My Games/OPHD/Saved Games. I've seen a couple of games use this approach and I find it easier to use. You could consider putting the saved games here in addition to telling people in the readme where they are if you don't want to fiddle with additional filesystem views in game. Sort of need to pick where to focus development time to finish one day...

**Comment by ldicker83** [comment]
I forgot that AppData is hidden by default.

Will have to look at the filesystem code and see if using Documents is a simple enough process. AppData is one of the preferred locations but based on the desktop application certification requirements (https://docs.microsoft.com/en-us/windows/win32/win_cert/certification-requirements-for-windows-desktop-apps) I think using the user's Documents folder would meet the requirements outlined in section 11. Settings can stay in AppData since those really shouldn't be modified directly by the user (later versions will give UI elements to change settings).
